File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text:")

letters = 0
words = 1
sentences = 0

for letter in text:
    if letter in ['.','!','?']:
        sentences += 1
    elif letter == ' ':
        words += 1
    elif letter.isalpha():
        letters +=1

print("Letters:", letters)
print("Words:", words)
l = (letters * 100) / words
s = (sentences * 100) / words
grade = int(((0.0588 * l) - (0.296 * s) - 15.8)+0.5)


if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print("Grade", grade)

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //variable declarations
    string texto = get_string("Text: ");
    int letras = 0;
    int palavras = 1;
    int frases = 0;
    //int letra = 0;
    int len = 0;

    //loops trough the text
    while (texto[len] != 0)
    {
        len++;
    }

    for (int i = 0; i < len; i++)
    {
        //if script encounter ponctuation marks, it adds a sentence
        if ((texto[i] == 46) || (texto[i] == 33) || (texto[i] == 63))
        {
            frases++;

        }

        //if it encounters a space, adds a word
        else if (texto[i] == 32)
        {
            palavras++;

        }

        //and add letters for all chars
        else if (((texto[i] >= 65) && (texto[i] <= 90)) || ((texto[i] >= 97) && (texto[i] <= 122)))
        {
            letras++;
        }

    }

    //doing some math stuff
    float l = ((letras * 100.00) / palavras);
    float s = ((frases * 100.00) / palavras);
    float index = ((0.0588 * l) - (0.296 * s) - 15.8);
    int grade = round(index);

    //showing the results
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}